grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -0,0 +1,17 @@
+package org.pih.warehouse.api
+
+import grails.converters.JSON
+import org.pih.warehouse.inventory.CycleCountCandidate
+import org.pih.warehouse.inventory.CycleCountCandidateFilterCommand
+import org.pih.warehouse.inventory.CycleCountService
+
+class CycleCountApiController {
+
+    CycleCountService cycleCountService
+
+    def getCandidates(CycleCountCandidateFilterCommand filterParams) {
+        List<CycleCountCandidate> candidates = cycleCountService.getCandidates(filterParams, params.facilityId)
+
+        render([data: candidates, totalCount: candidates.totalCount] as JSON)
```

Comment:
is the "totalCount" what will be used to display the number next to the ALL PRODUCTS tab?

If so, I'm not sure this will work if we apply any filters since that will shrink the result set of the query (and presumably the totalCount) but we want the number to continue to show for the total number of un-filtered elements.

Reply:
`totalCount` is meant for the pagination, so as you mentioned, whenever any filters are applied, the `totalCount` changes as well. I haven't thought about the number next to the "ALL PRODUCTS", but if you say, this should not change regardless the filters, we could just set it on frontend once (during the initial fetch, when none filters would be applied), and not update it whenever we refetch the data.